modin/core/storage_formats/base/query_compiler.py

```diff
@@ -2382,14 +2369,360 @@ def groupby_agg(
         elif drop and isinstance(by, type(self)):
             by = list(by.columns)
 
-        return GroupByDefault.register(pandas.core.groupby.DataFrameGroupBy.aggregate)(
+        return GroupByDefault.register(GroupByDefault.get_aggregation_method(how))(
             self,
             by=by,
-            is_multi_by=is_multi_by,
-            axis=axis,
             agg_func=agg_func,
-            groupby_args=groupby_kwargs,
-            agg_args=agg_kwargs,
+            axis=axis,
+            groupby_kwargs=groupby_kwargs,
+            agg_args=agg_args,
+            agg_kwargs=agg_kwargs,
+            drop=drop,
+        )
+
+    @doc_utils.doc_groupby_method(
+        action="compute the mean value", result="mean value", refer_to="mean"
+    )
+    def groupby_mean(
+        self,
+        by,
+        axis,
+        groupby_kwargs,
+        agg_args,
+        agg_kwargs,
+        drop=False,
+    ):
+        return self.groupby_agg(
+            by=by,
+            agg_func="mean",
+            axis=axis,
+            groupby_kwargs=groupby_kwargs,
+            agg_args=agg_args,
+            agg_kwargs=agg_kwargs,
+            drop=drop,
+        )
+
```

Comment:
I would like to generate those functions in a loop instead of copy-pasting them all over with changing one verb each time, something like
```python
for action, result, agg_func in (('compute the mean value', 'mean value', 'mean'), ('compute unbiased skew', 'unbiased skew', 'skew'), ...):
    @doc_utils.doc_groupby_method(action=action, result=result, refer_to=agg_func)
    def func(self, by, axis, groupby_kwargs, agg_args, agg_kwargs, drop=False):
        return self.groupby_agg(by=by, agg_func=agg_func, axis=axis, groupby_kwargs=groupby_kwargs, agg_args=agg_args, agg_kwargs=agg_kwargs, drop=drop)
    func.__name__ = f'groupby_{agg_func}'
    setattr(GroupByCompiler, func.__name__, func)
```

Reply:
`BaseQueryCompiler` is the place where we define QC API, I don't think that any kind of auto-generation can produce a readable interface.

Reply:
oh, if that's API definition I agree